### PR TITLE
Fix upsert freezes on rust client stress test

### PR DESCRIPTION
**Pull request by IvanPleshkov**
### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you lint your code locally using ``cargo fmt`` command prior to submission?
3. [ ] Have you checked your code using ```cargo clippy``` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by generall — APPROVED** [review]

**Comment by e-ivkov** [comment]
Does it make sense to submit an issue to `tokio` repository about slow performance compared to parking lot?

**Comment by IvanPleshkov** [comment]
> Does it make sense to submit an issue to `tokio` repository about slow performance compared to parking lot?

No because tokio has documentation with explanation when `parking_lot` mutex is preferable.

File: lib/collection/src/collection_manager/segments_updater.rs [hunk 1]
```diff
@@ -297,6 +297,7 @@ where
                 point.payload.as_ref(),
             )? as usize;
         }
+        RwLockWriteGuard::unlock_fair(write_segment);
```
Comment (agourlay) [inline lib/collection/src/collection_manager/segments_updater.rs:300]: why do weed this here?

**Comment by agourlay** [comment]
Here is the doc for history https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use

```
Contrary to popular belief, it is ok and often preferred to use the ordinary [Mutex](https://doc.rust-lang.org/nightly/std/sync/mutex/struct.Mutex.html) from the standard library in asynchronous code.

The feature that the async mutex offers over the blocking mutex is the ability to keep it locked across an .await point. This makes the async mutex more expensive than the blocking mutex, so the blocking mutex should be preferred in the cases where it can be used. The primary use case for the async mutex is to provide shared mutable access to IO resources such as a database connection. If the value behind the mutex is just data, it’s usually appropriate to use a blocking mutex such as the one in the standard library or [parking_lot](https://docs.rs/parking_lot).

Note that, although the compiler will not prevent the std Mutex from holding its guard across .await points in situations where the task is not movable between threads, this virtually never leads to correct concurrent code in practice as it can easily lead to deadlocks.

A common pattern is to wrap the Arc<Mutex<...>> in a struct that provides non-async methods for performing operations on the data within, and only lock the mutex inside these methods. The [mini-redis](https://github.com/tokio-rs/mini-redis/blob/master/src/db.rs) example provides an illustration of this pattern.

Additionally, when you do want shared access to an IO resource, it is often better to spawn a task to manage the IO resource, and to use message passing to communicate with that task.
```

**Pull request merged**